superbuild/CMakeLists.txt

```diff
@@ -373,9 +472,32 @@ if (APPLE AND CMAKE_SYSTEM_VERSION VERSION_GREATER_EQUAL 13)
     # or build and use it, otherwise a build error will occure (attempt to use
     # just built libcxxabi, which was not built).  An option to build seems to be
     # a better one.
+    set(LLVM_RUNTIMES
+        ${LLVM_RUNTIMES}$<SEMICOLON>libcxx$<SEMICOLON>libcxxabi
+        )
+endif()
+
+# LLVM_ENABLE_RUNTIMES is a proper place for compiler-rt and openmp not LLVM_ENABLE_PROJECTS.
+list(APPEND LLVM_COMMON_CMAKE_ARGS
+    -DLLVM_ENABLE_RUNTIMES=${LLVM_RUNTIMES}
+    )
+
+if (APPLE)
+    # macOS deployment target sets the minimim OS requirement to run the
+    # compiled program. All of the object files and static libraries passed to
+    # the linker need to be compiled for the same version of the macOS to
+    # enable resulting executable to be able to run on the older macOS. So we
+    # build LLVM for old macOS to enable ISPC builds targeting an old macOS.
     list(APPEND LLVM_COMMON_CMAKE_ARGS
-        -DLLVM_ENABLE_RUNTIMES=libcxx$<SEMICOLON>libcxxabi
+        -DCMAKE_OSX_DEPLOYMENT_TARGET=${MACOS_VERSION_MIN}
         )
+
+    if (MACOS_UNIVERSAL_BIN)
+        # macOS Universal Binaries is a fat binary for x86_64 and arm64.
+        list(APPEND LLVM_COMMON_CMAKE_ARGS
+            -DCMAKE_OSX_ARCHITECTURES=x86_64$<SEMICOLON>arm64
```

Comment:
Do we need quotes around `x86_64$<SEMICOLON>arm64`? I'm not sure about it, just checking.